src/plugins/section_category.ts

```diff
@@ -0,0 +1,257 @@
+/*!
+ * © 2024 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Model, SectionCategory } from '@manuscripts/json-schema'
+import { ManuscriptNode, schema } from '@manuscripts/transform'
+import { EditorState, Plugin, PluginKey } from 'prosemirror-state'
+import { Decoration, DecorationSet, EditorView } from 'prosemirror-view'
+
+import { sectionCategoryIcon } from '../assets'
+import { EditorProps } from '../configs/ManuscriptsEditor'
+import { PopperManager } from '../lib/popper'
+import {
+  getCategoryName,
+  isBackMatterSection,
+  isEditableSectionCategory,
+  isUnique,
+} from '../lib/section-categories'
+import { isChildOfNodeType } from '../lib/utils'
+
+export const sectionCategoryKey = new PluginKey<PluginState>('section-category')
+const popper = new PopperManager()
+
+export interface PluginState {
+  decorations: DecorationSet
+}
+
+export interface SectionCategoryProps extends EditorProps {
+  sectionCategories: SectionCategory[]
+}
+
+function changeNodeType(
```

Comment:
looks like that function changes category type and not node type. May be we can rename it to reflect more specifically what it does? Also not all nodes have category attributes, so it may be defined more specifically to the node type - i.e. node argument can be SectionNode.

Reply:
done